src/tools/r2rdump/R2RReader.cs

```diff
@@ -514,17 +537,46 @@ public int GetOffset(int rva)
         /// </summary>
         public static string GetTypeDefFullName(MetadataReader mdReader, TypeDefinitionHandle handle)
         {
-            TypeDefinition typeDef;
+            string typeNamespace = "";
             string typeStr = "";
             do
             {
-                typeDef = mdReader.GetTypeDefinition(handle);
-                typeStr = "." + mdReader.GetString(typeDef.Name) + typeStr;
-                handle = typeDef.GetDeclaringType();
+                try
+                {
+                    TypeDefinition typeDef = mdReader.GetTypeDefinition(handle);
```

Comment:
Just checking, do `GetTypeDefFullName` and `GetExportedTypeFullName` handle nested types properly?

Reply:
Do exported types have nested types? The ExportedType class doesn't provide GetDeclaringType() for getting the parent type.

GetTypeDefFullName does handle nested types. I'm not sure if there could be cases where TypeDefs are nested inside an exported type. That could cause an error

Reply:
Yes I think you're right. Looks good to me.